pkgs/development/tools/build-managers/bazel/default.nix

```diff
@@ -56,9 +56,396 @@ stdenv.mkDerivation rec {
     }
   '';
 
+  buildFile = writeText "BUILD" ''
+    package(default_visibility = ['//visibility:public'])
+
+    filegroup(name = "empty")
+
+    cc_toolchain_suite(
+      name = "nix",
+      toolchains = {
+        "darwin|compiler": ":nix_darwin_toolchain",
+        "k8|compiler": ":nix_linux_toolchain",
+      },
+    )
+
+    cc_toolchain(
+      name = "nix_darwin_toolchain",
+      all_files = ":osx_wrapper",
+      compiler_files = ":osx_wrapper",
+      cpu = "darwin",
+      dwp_files = ":empty",
+      dynamic_runtime_libs = [":empty"],
+      linker_files = ":osx_wrapper",
+      objcopy_files = ":empty",
+      static_runtime_libs = [":empty"],
+      strip_files = ":empty",
+      supports_param_files = 0,
+    )
+
+    cc_toolchain(
+      name = "nix_linux_toolchain",
+      all_files = ":empty",
+      compiler_files = ":empty",
+      cpu = "k8",
+      dwp_files = ":empty",
+      dynamic_runtime_libs = [":empty"],
+      linker_files = ":empty",
+      objcopy_files = ":empty",
+      static_runtime_libs = [":empty"],
+      strip_files = ":empty",
+      supports_param_files = 0,
+    )
+
+    filegroup(
+      name = "osx_wrapper",
+      srcs = ["osx_cc_wrapper.sh"],
+    )
+  '';
+
+  crosstoolFile = writeText "CROSSTOOL" (''
```

Comment:
This should use nix’ importing capabilities, since it’s a quite long file.
It consists mostly of “magic”, which is copied mostly verbatim as snippets from [here](https://github.com/bazelbuild/bazel/blob/0.13.0/tools/cpp/CROSSTOOL) as far as I can see.

It is also not clear at all from the nix file where the source comes from or what it does. Such a giant unknown code snippet should be documented extraordinarily well, and directly in the code (in order to not be “magic” to everybody but the commiter.)